api/database/seeders/DatabaseSeeder.php

```diff
@@ -79,22 +74,12 @@ public function run()
 
                 // are they a government user?
                 if (rand(0, 1)) {
-                    // government users have a current classification and expected classifications but no salary
-                    $classification = Classification::inRandomOrder()->limit(1)->pluck('id')->toArray();
-                    $user->expectedClassifications()->sync($classification);
-                    $user->expected_salary = [];
+                    // government users have a current classification
```

Comment:
Out of scope, but this part of the seeder looks odd. It seems as though we are just nulling the current classification for ~50% of users seeded. Wouldn't we also need to set the following:

```php
$user->is_gov_employee = false;
$user->department->null;
```

Reply:
I was also confused by the logic within the "are they a government user?" conditional. I will ask about it in today's stand-up.

Reply:
See 511f7815dcbfcb3c9eeb5fbf43d22685f8bede1c

Reply:
@petertgiles Do you have time to second check the state update? I paired up with Matt on this so we want to keep the review integrity in tact 😅

Reply:
@petertgiles do you mind reviewing this as well? 🙏

Reply:
Looks nice!  Having the boolean parameter on the state function is a little unusual.  I haven't seen it before but I don't think it's wrong.  Would be nice to have a little comment above each of the two factory calls (L63 and L102) to explain at a glance why there are two different blocks.

I don't love having the big copy-paste afterCreating callback duplicated but I think that's a symptom of needing to refactor that and not a problem with the work in this PR.  Calling the factory twice with different chains shouldn't be a problem.

Reply:
@petertgiles is this what you were looking for in terms of the comments: e6eb1a8c42238910f94883f105ce1e237a8ac2a6

I agree with the duplicate `afterCreating` code not being great, but I think part of a larger issue: https://github.com/GCTC-NTGC/gc-digital-talent/issues/5045.